### PR TITLE
Shorten zq and leibpilem1; add nn0onn

### PR DESCRIPTION
**Pull request by icecream17**
Two theorems I noticed could be shortened, before starting onto the generalize zsqrtelqelz and subtheorems

As a sidenote, dffltz can be improved by eldifsnneq (very recent) and ss2ralv (still in #3097). But I wonder if a temporary definition like ``F = ( n e. ( ZZ>= ` 3 ) , s e. ~P CC |-> { <. <. x , y >. , z >. | ( ( x ^ n ) + ( y ^ n ) ) = ( z ^ n ) } )`` would be worth it.

### CODE REVIEW

**Review by wlammen — APPROVED** [review]

File: set.mm
Comment (benjub) [inline set.mm:359485]: The proof as well as the usages seem to show that there is not much gain in having these two theorems bundled like that into one. Also, `-. N = 0` is implied. I think two theorems (and not merely lemmas) would be justified:
```
|- ( ( N e. NN0 /\ -. 2 || N ) -> N e. NN ) $=
|- ( ( N e. NN0 /\ -. 2 || N ) -> ( N - 1 ) / 2 ) e. NN0 ) $=
```
and maybe also
```
|- ( ( N e. NN0 /\ -. 2 || N ) -> ( N + 1 ) / 2 ) e. NN ) $=
```

**Comment by wlammen** [comment]
Is there a reason to keep this PR on hold any longer?  The two modified theorems are replacements for the better, both with shortened proofs, and none based on more axioms than the original.  While there might be headroom for further improvements or considerations in the big picture, I don't see why this cannot be postponed until this is merged first.

**Comment by icecream17** [comment]
(I was actually implementing benjub's first two theorems, although the point still stands)

Edit: Actually, given that the second theorem is just nn0oddm1d2 + biimpa, I'm only implementing nn0onn (this commit was made yesterday so it appears above wlammen's comment but it was actually published a few minutes after this comment)

**Comment by benjub** [comment]
Thanks. Now that we have both ~nn0onn and ~nn0oddm1d2, and looking at the two usages of ~leibpilem1, I think that ~leibpilem1 can be removed (i.e., in its two usages, replace it with ~nn0onn or ~nn0oddm1d2 with possibly some glue).

File: set.mm
Comment (benjub) [inline set.mm:175589]: for consistency with the following comments:
>  $( 2 divides 0.  That means 0 is even.

File: set.mm
Comment (benjub) [inline set.mm:175607]: > $( 2 divides 2. That means 2 is even.

File: set.mm
Comment (benjub) [inline set.mm:175612]:
```suggestion
  $( 2 does not divide 3. That means 3 is odd.  (Contributed by AV,
```

File: set.mm
Comment (benjub) [inline set.mm:175618]: $( 2 divides 4. That means 4 is even.

File: set.mm
Comment (benjub) [inline set.mm:175624]: > $( An integer which is divisible by 4 is divisible by 2, that is, is even.

**Comment by icecream17** [comment]
Would there be leibpilem2OLD and leibpiOLD?

This could be more justified if I rewrite leibpi* to not include the redundant `k = 0`.
Additionally, both times leibpilem1 is used, it's really used to prove 
`(𝑘 ∈ NN0 → if((𝑘 = 0 ∨ 2 ∥ 𝑘), 0, ((-1↑((𝑘 − 1) / 2)) / 𝑘)) ∈ ℂ)`, which suggests a completely separate lemma, although some of the intermediate steps are used for other closures... But this pr is already getting much bigger than expected.

**Review by avekens — APPROVED** [review]
For me, this PR is fine. The revision of the comments as proposed by @benjub would be OK for me, too.

**Comment by benjub** [comment]
> For me, this PR is fine. The revision of the comments as proposed by @benjub would be OK for me, too.

Fine for me. I made some suggestions but nothing is blocking.

File: set.mm
Comment (icecream17) [reply set.mm:175589]: Alright these suggestions should be applied now

**Comment by jkingdon** [comment]
There seems to be a consensus this is ready to merge. Feel free to make follow up pull requests if there are other things people want to change.

**Pull request merged**
